shell/src/main/java/alluxio/cli/fs/command/CpCommand.java

```diff
@@ -140,21 +166,26 @@ public CopyThreadPoolExecutor(int threads, PrintStream out, AlluxioURI path) {
     }
 
     /**
-     * Prints out the message into the specified output stream in a new line, waiting if necessary
-     * for the internal message queue to have available space.
+     * Sends a message to the thread pool, if the message is of type String, then it will be
+     * displayed in stdout, if the message is of type Exception, then it will be displayed in
+     * stderr and collected into a list of exceptions, other types of messages will be ignored,
+     * this method will wait if necessary for the internal message queue to have available space.
      *
-     * @param message the message, must not be "#"
-     * @throws InterruptedException if interrupted while waiting
+     * @param message the message, either {@link String} or {@link Exception}
```

Comment:
or Object when done

Reply:
the MESSAGE_DONE object is internal, for user of this method, only `String` and `Exception` are allowed.

Reply:
I refactored the `sendMessage` API so that the messages are in a better controlled format.